src/main/java/org/tron/common/runtime/vm/PrecompiledContracts.java

```diff
@@ -973,11 +993,18 @@ public long getEnergyForData(byte[] data) {
 
       final List<Actuator> actuatorList = ActuatorFactory
           .createActuator(trx, getDeposit().getDbManager());
+      Optional<List<Actuator>> optionalList = Optional.ofNullable(actuatorList);
+      ProposalApproveActuator proposalApproveActuator;
       try {
-        actuatorList.get(0).validate();
-        actuatorList.get(0).execute(getResult().getRet());
-        getDeposit()
-            .syncCacheFromAccountStore(ByteString.copyFrom(getCallerAddress()).toByteArray());
+        if (!optionalList.isPresent() || optionalList.get().isEmpty()){
+          throw new ContractExeException("can't initiate ProposalApproveActuator for precompiled vm method");
+        }
+        else{
+          proposalApproveActuator = (ProposalApproveActuator)optionalList.get().stream().findFirst().get();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS3655))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1420&issues=AWWTWS-S7khxiKmGgQPj&open=AWWTWS-S7khxiKmGgQPj)

[ISSUE_KEY]: AWWTWS-S7khxiKmGgQPj